agent-inject/metrics.go

```diff
@@ -0,0 +1,73 @@
+package agent_inject
+
+import (
+	"slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	admissionv1 "k8s.io/api/admission/v1"
+)
+
+const (
+	metricsNamespace         = "vault"
+	metricsSubsystem         = "agent_injector"
+	metricsLabelNamespace    = "namespace"
+	metricsLabelType         = "injection_type"
+	metricsLabelTypeDefault  = "sidecar"
+	metricsLabelTypeInitOnly = "init_only"
```

Comment:
The injector adds an init container and a sidecar container by default, so default = "sidecar" doesn't sound right to me. I'd say there are three injection types: init only, sidecar only, and both. In the future we'll hopefully support [native sidecars](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/) as an injection type.